src/services_models/services_constructor.jl

```diff
@@ -66,10 +66,8 @@ function construct_service!(
             contributing_devices =
                 [d for d in contributing_devices if typeof(d) ∉ incompatible_device_types]
         end
-        if isempty(contributing_devices)
-            @warn("The contributing devices for service $(PSY.get_name(service)) is empty, cinsider removing the service from the system")
-            continue
-        end
+        #Services without contributing devices should have been filtered out in the validation
```

Comment:
You might be missing my point.  The statement just above this one performs another filter on contributing_devices.  The code is now correct, but your comment is inconsistent with that.

Reply:
I don't follow, if contributing devices for the service is empty the filtering didn't work. The above code in lines 63 - 66 all it does is find the contributing devices for the particular service.

Reply:
It seems possible that `[d for d in contributing_devices if typeof(d) ∉ incompatible_device_types]` could return an empty array and then cause your assert to trip.  Maybe that it not possible based on conditions unbeknownst to me; that's just how the code reads.